hello/hello_async_activity_completion.py

```diff
@@ -14,42 +17,39 @@ class ComposeGreetingInput:
 
 
 class GreetingComposer:
-    def __init__(self, client: Client) -> None:
+    def __init__(self, client: Client, loop: asyncio.AbstractEventLoop) -> None:
         self.client = client
+        self.loop = loop
 
     @activity.defn
-    async def compose_greeting(self, input: ComposeGreetingInput) -> str:
-        # Schedule a task to complete this asynchronously. This could be done in
+    def compose_greeting(self, input: ComposeGreetingInput) -> str:
+        # Make a thread to complete this externally. This could be done in
         # a completely different process or system.
         print("Completing activity asynchronously")
-        # Tasks stored by asyncio are weak references and therefore can get GC'd
-        # which can cause warnings like "Task was destroyed but it is pending!".
-        # So we store the tasks ourselves.
-        # See https://docs.python.org/3/library/asyncio-task.html#creating-tasks,
-        # https://bugs.python.org/issue21163 and others.
-        _ = asyncio.create_task(
-            self.complete_greeting(activity.info().task_token, input)
-        )
+        Thread(
+            target=self.complete_greeting,
+            args=(activity.info().task_token, input),
+        ).start()
 
         # Raise the complete-async error which will complete this function but
         # does not consider the activity complete from the workflow perspective
         activity.raise_complete_async()
 
-    async def complete_greeting(
-        self, task_token: bytes, input: ComposeGreetingInput
-    ) -> None:
+    def complete_greeting(self, task_token: bytes, input: ComposeGreetingInput) -> None:
         # Let's wait three seconds, heartbeating each second. Note, heartbeating
         # during async activity completion is done via the client directly. It
         # is often important to heartbeat so the server can know when an
         # activity has crashed.
         handle = self.client.get_async_activity_handle(task_token=task_token)
         for _ in range(0, 3):
             print("Waiting one second...")
-            await handle.heartbeat()
-            await asyncio.sleep(1)
+            asyncio.run_coroutine_threadsafe(handle.heartbeat(), self.loop).result()
```

Comment:
Maybe an explainer on why we're using coroutine_threadsafe?

Reply:
Also saw a comment today about how the activity reacts to cancellations.

It's not the heartbeat that throws the cancellation, it's done via asyncio cancellations. 

Not sure if that's out of the scope of this PR or sample